numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,94 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
 
     return view
 
+def sliding_window_view(x, shape, step=None):
+    """
+    Create rolling window views of the 2D array with the given shape.
+
+    For some cases, there may be more efficient approaches
+
+    Parameters
+    ----------
+    x : ndarray
+        Array to create rolling window views.
+    shape : sequence of int
+        The shape of the sliding window.
+    step: sequence of int, optional
+        The sliding step of view window for each dimension of input. Defaults to 1 on all dimensions.
+
+    Returns
+    -------
+    view : ndarray
+
+    Examples
+    --------
+    >>> i, j = np.ogrid[:3,:4]
+    >>> x = 10*i + j
+    >>> shape = (2,2)
+    >>> np.lib.stride_tricks.sliding_window_view(x, shape)
+    array([[[[ 0,  1],
+             [10, 11]],
+
+            [[ 1,  2],
+             [11, 12]],
+
+            [[ 2,  3],
+             [12, 13]]],
+
+
+           [[[10, 11],
+             [20, 21]],
+
+            [[11, 12],
+             [21, 22]],
+
+            [[12, 13],
+             [22, 23]]]])
+
+
+    >>> i, j = np.ogrid[:3,:4]
+    >>> x = 10*i + j
+    >>> shape = (2,2)
+    >>> step = (1,2)
+    >>> sliding_window_view(x, shape, step)
+    array([[[[ 0,  1],
+             [10, 11]],
+
+            [[ 2,  3],
+             [12, 13]]],
+
+
+           [[[10, 11],
+             [20, 21]],
+
+            [[12, 13],
+             [22, 23]]]])
+
+    """
+    shape = np.array(shape, np.int)
+
+    if step is None:
+        step = np.ones(len(x.shape), np.int)
+    else:
+        step = np.array(step, np.int)
+
+    if len(x.shape) != len(shape) or len(shape) != len(step):
+        raise ValueError("view shape or step dimension doesn't match with input array")
+    if np.any(shape <= 0) and np.any(step <= 0):
+        raise ValueError('view shape or step cannot contain non-positive value')
+
+    o = (np.array(x.shape)  - shape) // step + 1 # output shape
+    if np.any(o <= 0):
+        raise ValueError('view shape cannot larger than input array shape')
+
+    strides = x.strides
+    view_strides = strides * step
+
+    view_shape = np.concatenate((o, shape), axis=0)
+    view_strides = np.concatenate((view_strides, strides), axis=0)
+    view = np.lib.stride_tricks.as_strided(x, view_shape, view_strides)
+    return np.squeeze(view)
```

Comment:
This is a bad idea, as it makes it impossible to ask for a length-1 window.

Reply:
fixed